### ISSUE TITLE
Check PSNR and SSIM

### ISSUE THREAD

**Issue by leviethung2103** [HyeongseokSon1/SRFeat#7]
Hello author HyeongseokSon, 

**I've used your pretrained model to evaluate the Set5 and Set14 dataset. However, the results I got are not identical to your paper.** 

I used your SRFeat_full.npz file to evaluate Set 5 dataset. I get all the images that are created from model, then I do the comparison with  HR images of Set5. 

In my code for evaluation (written in Matlab):
Firstly, I converted all images into YCbCr channels, get the Y channel only, shave the border 4 pixels and use PSNR/SSIM functions of Matlab.

I finally get the results 
SSIM: 0.844517
PNSR: 29.990681
for Set 5 

and 
SSIM: 0.711998
PNSR: 26.563117
for Set 14. 

Could you provide the code for your evaluation sets or I will provide my code for you to double check if there is any wrong with my code. 

Thank you a lot,

**Comment by leviethung2103** [comment]
[result.docx](https://github.com/HyeongseokSon1/SRFeat/files/2670811/result.docx)

**Comment by HyeongseokSon1** [comment]
First of all, please note that our main paper evaluated PSNR/SSIM of pre-trained model, trained by only MSE loss.
And we reported only PSNR of full model, trained by GAN loss in supplementary material.
The values are 30.03 for Set 5 and 26.73 for Set 14

Our test code is based on the test code of EDSR paper (https://github.com/limbee/NTIRE2017/blob/master/demo/evaluation.m).
And I found your code is slightly wrong.
Because indices in Matlab start from 1, you should change the code
from
    % Shave border
    sr_image = sr_image(real_shave : H +1 - real_shave, real_shave :W + 1 -real_shave);
    hr_image = hr_image(real_shave : H +1 - real_shave, real_shave :W + 1 -real_shave);
to
    % Shave border
    sr_image = sr_image(real_shave  + 1 : H - real_shave, real_shave + 1 : W - real_shave);
    hr_image = hr_image(real_shave + 1 : H - real_shave, real_shave + 1 : W - real_shave);

Because the values are slightly different, I think that it can be the reason.

**Comment by leviethung2103** [comment]
Thank you so much for your enthusiastic help. 

I realize that the shave should be 4 and you are so right about the code Shave border. I tested with Set 5 and results is the same as paper.

However, just only case of Set14, my result is lower, so I guess there is a problem with images in Set14. In particularly, there is a file 8 bit named "bridge.png" that leads to my worse result.

Have you in that situation before when testing with Set14 ? 

I really appreciate your help. 
![untitled](https://user-images.githubusercontent.com/16741872/49856433-b98e6a80-fe2a-11e8-8c8d-048b372dadd4.png)

**Comment by HyeongseokSon1** [comment]
In your code, you handle gray-scale case (C==1) only for HR images.
I think that you should not use rgb2ycbcr function for SR if HR is gray-scale.

**Comment by leviethung2103** [comment]
You're so right I have a mistake in the previous code. Thank you for noticing me about that.
However, SR images from model have 3 channels. There are two possibilites I am concerning on the case of grayscale image.


** 1/ Don't use rgb2ycbcr on SR, pickup the first channel of SR and compare directly to HR.
**2/ Duplicate 1 channel -> 3 channels of HR image, then use rgb2ycbcr for both SR and HR. 

That's weird because when I am trying the second methods, the produced result is the same as you mentioned above as well as in paper. or maybe it's a correct way [?]

Here is my code and the result. 

I will try the code you sent me above and try it tomorrow. Anyway, thank you for spending your precious time to answer my questions. 

![untitled](https://user-images.githubusercontent.com/16741872/49860916-b482e880-fe35-11e8-9aeb-32d5bdf3232e.png)
